framework/src/loops/NonlinearThread.C

```diff
@@ -42,12 +45,20 @@ NonlinearThread::NonlinearThread(NonlinearThread & x, Threads::split split)
     _dg_kernels(x._dg_kernels),
     _interface_kernels(x._interface_kernels),
     _kernels(x._kernels),
-    _tag_kernels(x._tag_kernels)
+    _tag_kernels(x._tag_kernels),
+    _has_active_objects(x._has_active_objects)
 {
 }
 
 NonlinearThread::~NonlinearThread() {}
 
+void
+NonlinearThread::operator()(const ConstElemRange & range, bool bypass_threading)
+{
+  if (_has_active_objects)
+    ThreadedElementLoop<ConstElemRange>::operator()(range, bypass_threading);
```

Comment:
are you going to be ok skipping the jacobian loop?

It s not a NonLinearThread  inherited class for now, but it will be soon

Reply:
I was checking that. Looks like it is good. `ComputeJacobianThread` has only these four warehouses as well. And the skipping should be logically correct.

Reply:
same warehouses right now between Jacobian and Residual. But yes